coderadar-ui/src/app/service/project.service.ts

```diff
@@ -214,7 +203,7 @@ export class ProjectService {
    * @param branch The branch to use for getting the commits.
    */
   public getCommitsForContributor(id: number, branch: string, email: string): Promise<HttpResponse<Commit[]>> {
-    return this.httpClient.get<Commit[]>(this.apiURL + 'projects/' + id + '/' + branch + '/commits?email='+email,
+    return this.httpClient.get<Commit[]>(this.apiURL + 'projects/' + id + '/' + branch + '/commits?email=' + email.replace('+', '%2B'),
```

Comment:
Why do you replace all '+' in generated url with '%2B'? '+' has to be encoded, but I don't see any place where a '+' could be added to the url.

Also, if the urls have multiple parameter, we may consider using [string literals](https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/template_strings) to increase readability.

Reply:
The + sign is an edge case here. Some emails (those generate by GitHub for example) contain a plus sign. I agree about the string literals.